google-cloud-spanner/lib/google/cloud/spanner/pool.rb

```diff
@@ -241,15 +239,24 @@ def shutdown
         end
 
         def new_session!
-          session = @client.create_new_session
-
           @mutex.synchronize do
-            # don't add if the pool is closed
-            return session.release! if @closed
```

Comment:
This is removing the check for the client being closed. It is possible that client is blocking on a new session and when it gets unblocked the client is closed. In that case, should we not create a new session and raise here?